test/test_ops.py

```diff
@@ -237,89 +262,132 @@ def _ref_test_helper(
                 )
             ):
                 continue
-            with ctx():
-                ref_result = op(sample.input, *sample.args, **sample.kwargs)
-            torch_result = op.torch_opinfo(sample.input, *sample.args, **sample.kwargs)
-
-            for a, b in zip(tree_flatten(ref_result)[0], tree_flatten(torch_result)[0]):
-                if isinstance(a, torch.Tensor) or isinstance(b, torch.Tensor):
-                    prims.utils.compare_tensor_meta(a, b)
-                    if getattr(op, 'validate_view_consistency', True) and not skip_view_consistency:
-                        msg = (f"The torch implementation {'returns' if b._is_view() else 'does not return'} "
-                               f"a view, while the reference {'does' if a._is_view() else 'does not'}")
-                        self.assertEqual(a._is_view(), b._is_view(), msg)
-
-            # Computes the dtype the more precise computatino would occur in
-            precise_dtype = torch.bool
-            if prims.utils.is_integer_dtype(dtype):
-                # Note: bool and integer dtypes do not have more
-                # precise dtypes -- they simply must be close
-                precise_dtype = dtype
-            if prims.utils.is_float_dtype(dtype):
-                precise_dtype = torch.double
-            if prims.utils.is_complex_dtype(dtype):
-                precise_dtype = torch.cdouble
-
-            # Checks if the results are close
-            try:
-                self.assertEqual(
-                    ref_result,
-                    torch_result,
-                    exact_stride=False,
-                    exact_device=True,
-                    exact_layout=True,
-                    exact_is_coalesced=True,
-                )
-            except AssertionError as e:
-                # Raises the error if the precise dtype comparison wouldn't be
-                # different
-                if dtype is precise_dtype:
-                    raise e
 
-                ex = e
+            # Check op behavior and API compatibility inside and outside ctx.
```

Comment:
So the current test compares the `torch.foo` vs. `torch._refs.foo`, right? And this extends testing to include:

1.  redirecting from `torch.foo` to `torch._refs.foo` via `torch_function`
2. testing alias consistency

Today in `test_ops.py` we test alias consistency in a very time-consuming way here:

https://github.com/pytorch/pytorch/blob/70cce9f8d1a099af8b017d7263897d3ca2fb9fe6/test/test_ops.py#L867

When really we'd like to test aliases by just asking if `foo_alias is foo`. Of course, as we've discussed briefly in these notes there are also "near aliases" that wouldn't satisfy the above test, but we should probably just treat those as separate operations or fix them so they're actual aliases. Note that for `torch.*` aliases it's not true that `torch.foo_alias is foo` because of how we decided to create alias for operations (they have autogenerated bindings from native_functions.yaml), but Python references don't use yaml or C++, and so could be defined to satisfy the `is` expression, as this PR does.

Anyway, back to these text extensions. The first part of the test that validates the `torch_function` mapping is correct seems more like we're testing that mapping than anything, and I imagine that's already done or best done elsewhere? Further, shouldn't we validate that mapping by ensuring the mapped-to operation is correct, rather than using blackbox testing? 

As for aliases, a separate test for Python Reference Alias Consistency is fine. It could start by asking if `alias_foo is foo` and if that's not true then it could assume that the alias is a "near alias" that should be blackbox tested for input->output consistency. 

How does that sound?